R-packages/covidcast/R/covidcast.R

```diff
@@ -391,23 +391,20 @@ plot.covidcast_signal <- function(x, plot_type = c("choro", "bubble", "line"),
                                   alpha = 0.5, direction = FALSE,
                                   dir_col = c("#6F9CC6", "#E4E4E4", "#C56B59"),
                                   bubble_col = "purple", num_bins = 8,
-                                  line_col = 1:6,
-                                  line_type = rep(1:6, each = length(line_col)),
                                   title = NULL, choro_params = list(),
                                   bubble_params = list(), line_params = list(),
                                   ...) {
   plot_type <- match.arg(plot_type)
   x <- latest_issue(x)
 
-  # Set range, if we need to (to mean +/- 3 standard deviations, from metadata)
-  if (is.null(range)) {
-    if (is.null(attributes(x)$metadata)) {
-      warn(
-        paste0("Metadata for signal mean and standard deviation not available; ",
-               "defaulting to observed mean and standard deviation to set plot range."),
-        class = "covidcast_plot_meta_not_found"
-      )
-
+  # For the maps, set range, if we need to (to mean +/- 3 standard deviations,
+  # from metadata) 
+  if (is.null(range) && plot_type == "choro" || plot_type == "bubble") {
```

Comment:
`&&` has higher precedence than `||`, so this is

```r
if ((is.null(range) && plot_type == "choro") || plot_type == "bubble") {
```

But you probably want 

```suggestion
  if (is.null(range) && (plot_type == "choro" || plot_type == "bubble")) {
```

otherwise `range` will always be ignored for bubble plots.

Reply:
Good catch!